File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) <= 1:
            return len(s)
        len_long = 0
        for i in range(len(s)):
            a = i
            curr_cnt = {}
            while a < len(s):
                curr_cnt.setdefault(s[a], 0)
                curr_cnt[s[a]] += 1
                if max(curr_cnt.values()) > 1:
                    break
                else:
                    a += 1
        
            len_long = max(len_long, a-i)
        
        return len_long

File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int maxLength = 0;
        for (int i=0; i < s.length(); i++) {
            Set<Character> set = new HashSet<>();
            for(int j=i; j < s.length(); j++) {
                if(!set.contains(s.charAt(j))) {
                    maxLength = Math.max(maxLength, j-i+1);
                    set.add(s.charAt(j));
                } else {
                    break;
                }
            }
        }
        return maxLength;
    }
}